packages/databases-collections-list/src/namespace-card.tsx

```diff
@@ -0,0 +1,311 @@
+        onDefaultAction(evt);
+      }
+    },
+    [onDefaultAction]
+  );
+
+  return { onMouseDown, onClick, onKeyDown };
+}
+
+const cardTitleGroup = css({
+  display: 'flex',
+  alignItems: 'center',
+  gap: spacing[3],
+  marginBottom: spacing[2],
+});
+
+const CardTitleGroup: React.FunctionComponent = ({ children }) => {
+  return <div className={cardTitleGroup}>{children}</div>;
+};
+
+const cardNameWrapper = css({
+  // Workaround for uncollapsible text in flex children
+  minWidth: 0,
+});
+
+const cardName = css({
+  overflow: 'hidden',
+  whiteSpace: 'nowrap',
+  textOverflow: 'ellipsis',
+  // To make container 28px to match leafygreen buttons
+  paddingTop: 2,
+  paddingBottom: 2,
+  // Because leafygreen
+  color: `${uiColors.green.dark1} !important`,
+  // TS is very confused if fontWeight is not a number even though it's a valid
+  // CSS value
+  fontWeight: '600 !important' as unknown as number,
+});
+
+const CardName: React.FunctionComponent<{ children: string }> = ({
+  children,
+}) => {
+  return (
+    <div title={children} className={cardNameWrapper}>
+      <Subtitle as="div" className={cardName}>
+        {children}
+      </Subtitle>
+    </div>
+  );
+};
+
+const cardActionContainer = css({
+  marginLeft: 'auto',
+  flex: 'none',
+});
+
+const CardActionContainer: React.FunctionComponent = ({ children }) => {
+  return <div className={cardActionContainer}>{children}</div>;
+};
+
+const cardBadges = css({
+  display: 'flex',
+  gap: spacing[2],
+  // Preserving space for when cards with and without badges are mixed in a
+  // single row
+  minHeight: 20,
+});
+
+const CardBadges: React.FunctionComponent = ({ children }) => {
+  return <div className={cardBadges}>{children}</div>;
+};
+
+const cardBadge = css({
+  gap: spacing[1],
+});
+
+const cardBadgeLabel = css({});
+
+export type BadgeProp = {
+  name: string;
+  variant?: BadgeVariant;
+  icon?: IconGlyph;
+  hint?: React.ReactNode;
+};
+
+const CardBadge: React.FunctionComponent<BadgeProp> = ({
+  name,
+  icon,
+  variant,
+  hint,
+}) => {
+  const badge = useCallback(
+    ({ className, children, ...props } = {}) => {
+      return (
+        <Badge
+          className={cx(cardBadge, className)}
+          variant={variant}
+          {...props}
+        >
+          {icon && <Icon size="small" glyph={icon}></Icon>}
+          <span className={cardBadgeLabel}>{name}</span>
+          {/* Tooltip will be rendered here */}
+          {children}
+        </Badge>
+      );
+    },
+    [icon, name, variant]
+  );
+
+  if (hint) {
+    return (
+      <Tooltip delay={500} trigger={badge}>
+        {hint}
+      </Tooltip>
+    );
+  }
+
+  return badge();
+};
+
+const card = css({
+  padding: spacing[3],
+});
+
+export type DataProp = {
+  label: React.ReactNode;
+  value: React.ReactNode;
+  hint?: React.ReactNode;
+};
+
+export type NamespaceItemCardProps = {
+  id: string;
+  type: ItemType;
+  viewType: ViewType;
+  name: string;
+  status: string;
```

Comment:
How do you feel about using enums instead of strings for things like status?

Something like:
```ts
enum NamespaceStatus {
  Initial = 'Initial'
  ....
}
```

So it can be used like `status={NamespaceStatus.Initial}`

And then
`status: NamespaceStatus`

It's a bit more passing around, but I think it might make the statuses more type defined?

Reply:
Good catch, this does need better typing that I'll add, but I prefer not to do it with enums if that's fine (for reason you described basically and Anna mentioned a few times in other PRs)

Reply:
Done in 45c676ebb33f00daf4b10e4742794018eab575f4